### ISSUE TITLE
Output discrepancy between PyTorch and ONNX Runtime when using `dynamo=True` with `model.eval()` in `torch.onnx.export`

### ISSUE THREAD

**Issue by xyecoding** [pytorch/pytorch#168224]
### 🐛 Describe the bug

When exporting a PyTorch model in evaluation mode using the new Dynamo-based ONNX exporter (dynamo=True), the serialized ONNX model produces outputs that differ significantly from the original PyTorch model's outputs. The same code works correctly when using the legacy exporter (dynamo=False).

```python
import torch
from torchvision import models


torch_model = models.resnet50(pretrained=True).eval()
example_inputs = (torch.randn(1, 3, 224, 224),)
torch.onnx.export(
    torch_model,
    example_inputs,
    "image_classifier_model.onnx",
    dynamo=True,
)


import onnxruntime

onnx_inputs = [tensor.numpy(force=True) for tensor in example_inputs]
print(f"Input length: {len(onnx_inputs)}")
print(f"Sample input: {onnx_inputs}")

ort_session = onnxruntime.InferenceSession(
    "./image_classifier_model.onnx",
    providers=["CPUExecutionProvider"],
)

onnxruntime_input = {
    input_arg.name: input_value
    for input_arg, input_value in zip(ort_session.get_inputs(), onnx_inputs)
}

onnxruntime_outputs = ort_session.run(None, onnxruntime_input)[0]
torch_outputs = torch_model(*example_inputs)
assert len(torch_outputs) == len(onnxruntime_outputs)
for torch_output, onnxruntime_output in zip(torch_outputs, onnxruntime_outputs):
    torch.testing.assert_close(torch_output, torch.tensor(onnxruntime_output))
print("PyTorch and ONNX Runtime output matched!")
print(f"Output length: {len(onnxruntime_outputs)}")
print(f"Sample output: {onnxruntime_outputs}")
```

### Versions

PyTorch Version: 2.8.0+cu128
Torchvision Version: 0.23.0+cu128
onnxruntime Version: 1.23.2
Python Version: 3.13.0

cc @justinchuby @titaiwangms

**Comment by titaiwangms** [comment]
Please try nightly.
